src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -1037,15 +1037,15 @@ private List<string> GetUndeclaredProjects(MSBuild msbuildTask)
 
             var projectReferenceItems = _buildRequestEntry.RequestConfiguration.Project.GetItems(ItemTypeNames.ProjectReference);
 
-            var declaredProjects = new HashSet<string>(projectReferenceItems.Count);
+            var declaredProjects = new HashSet<string>(projectReferenceItems.Count + 1, StringComparer.OrdinalIgnoreCase);
```

Comment:
> I'm not sure about OrdinalIgnoreCase here. People can (and sometimes do) have two projects identical but for casing, and this would break that scenario. Can you provide a compelling example for which we need OrdinalIgnoreCase here?

@Forgind the issue was that the current project somehow ended up with a different case than what it was testing against.  So the it errored saying f:\foo.csproj is not allowed to call F:\foo.csproj